test/add-on_auto_test.rb

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+require "yast"
+require_relative "../src/clients/add-on_auto"
```

Comment:
Beware that this actually executes some code! The old clients converted from the YCP language run the `#main` method just after the class definition. See https://github.com/dgdavid/yast-add-on/blob/8194f919c42c6d2efc60dde2f4fc67b50407b969/src/clients/add-on_auto.rb#L330

To make this class testable move the class definition to a `lib/src` file (without that `#main` call at the end) and `require` it in the original place. Then you can use it in tests.

See https://github.com/yast/yast-add-on/blob/master/src/lib/add-on/clients/inst_add-on.rb and https://github.com/yast/yast-add-on/blob/master/src/clients/inst_add-on.rb files.

Reply:
You might also consider inheriting from https://github.com/yast/yast-yast2/blob/master/library/general/src/lib/installation/auto_client.rb. However, that's out of the scope of this PR, but have it in mind if you are refactoring this class in another branch.

Reply:
@lslezak @imobachgs 

Thank you guys :innocent:

 I'll to follow the @lslezak advices for this PR and I also will take into account the link provided by @imobachgs for the refactorization branch. In fact, I was starting to split the big #main `if` statement into independent methods, which it seem to be a similar approach to the already done in the `AutoClient#run`. Quite useful :+1: